packages/types/config/index.d.ts

```diff
@@ -45,6 +45,8 @@ export interface Configuration extends Record<string, any> {
   modules?: NuxtConfigurationModule[]
   modulesDir?: string[]
   plugins?: NuxtConfigurationPlugin[]
+  privateRuntimeConfig?: Record<string, any> | ((env: NuxtConfigurationEnv) => Record<string, any>)
+  publicRuntimeConfig?: Record<string, any> | ((env: NuxtConfigurationEnv) => Record<string, any>)
```

Comment:
I don't fully get the function form idea. It can only takes objects with string values? Does it not work with "any" type values?

Reply:
I think @pi0 can confirm that `env` parameter will be direct values from `dotenv` which are only `string`

Reply:
Yeah. It is an internal `_env` object. The function is not really suggested. The only reason was that some requested supporting `--no-processenv` in that case, only way to use loaded env would be using function format.